packages/Combobox/src/combobox.hook.tsx

```diff
@@ -0,0 +1,455 @@
+import { useClickable } from "@chakra-ui/clickable"
+import { useDescendant, useDescendants } from "@chakra-ui/descendant"
+import { useDisclosure, useId, useIds, useUpdateEffect } from "@chakra-ui/hooks"
+import { UsePopperProps } from "@chakra-ui/popper"
+import {
+  callAllHandlers,
+  createOnKeyDown,
+  getNextIndex,
+  //getNextItemFromSearch,
+  getPrevIndex,
+  focus,
+  createContext,
+} from "@chakra-ui/utils"
+import { InputProps } from "@chakra-ui/input"
+import * as React from "react"
+
+const [ComboboxContextProvider, useComboboxContext] = createContext<
+  UseComboboxReturn
+>({
+  strict: false,
+  name: "ComboboxContext",
+})
+
+export { ComboboxContextProvider, useComboboxContext }
+
+export interface UseComboboxProps extends UsePopperProps {
+  /**
```

Comment:
What should be the API for a Combobox component.  The reason why I started building one is because I need a Google Places Autocomplete Component.  In my case the list of items will be loaded on demand.  I will need an `onChange` event and  a `isLoading` prop to render the loading icon while it waits for the server response.  I am also thinking about a `multi` prop, where multiple items can be selected.  
But what else?